### PR TITLE
CI: Back up and restore a VM.

### PR DESCRIPTION
**Pull request by mrnold**
This pull request adds an actual backup and restore of a running VM. This includes some rework of the existing backup and restore test code, so that different parts of the sequence could be specialized for VM purposes. The result is a little bit awkward, but it makes almost no functional changes to existing tests.

### CODE REVIEW

**Comment by kaovilai** [comment]
/hold
> re:Do not merge this yet

**Comment by mrnold** [comment]
Okay, the VM backup and restore test has worked successfully at least once. This is at least worth having @mateusoliveira43 and @mpryc take a look at it.

**Comment by mrnold** [comment]
Looks like failures in "Mongo application DATAMOVER" instead of "MySQL application two Vol CSI" this time, maybe one more retest for entertainment.

**Comment by mateusoliveira43** [comment]
@weshayutin this resolves https://issues.redhat.com/browse/OADP-2982 ?

**Comment by weshayutin** [comment]
@mateusoliveira43 it only partially resolves OADP-2982.    I'll review how to associate a pr w/ a jira again w/ @mrnold .  It's a just a jira task and for CI.

File: tests/e2e/lib/backup.go
Comment (mateusoliveira43) [inline tests/e2e/lib/backup.go:29]: maintainers, I never used this flag. This is needed if the PV is not in the same namespace as one of in `IncludedNamespaces`?

File: tests/e2e/lib/restore.go
Comment (mateusoliveira43) [inline tests/e2e/lib/restore.go:27]: maintainers, what this option does with the restore?

File: tests/e2e/backup_restore_suite_test.go
Comment (mateusoliveira43) [inline tests/e2e/backup_restore_suite_test.go:1]: I would suggest breaking this PR in 2

- add backup/restore virt
- remove duplications from backup/restore and must-gather tests

File: tests/e2e/lib/backup.go
Comment (sseago) [reply tests/e2e/lib/backup.go:29]: PVs are cluster-scoped and are not in any namespace. PVCs that are not in an included namespace will be excluded from backup.
This flag tells Velero to snapshot PVCs that are 1) included in the backup, and 2) not backed up using fs-backup. If this is false, then non-fs-backup PVCs will not be snapshotted.

File: tests/e2e/lib/restore.go
Comment (sseago) [reply tests/e2e/lib/restore.go:27]: If I'm remembering correctly, this the restore equivalent to snapshotVolumes -- @shubham-pampattiwar is this correct?
Assuming I'm not misremembering, this means, similar to above, if it's true then we restore the PVC content from snapshot if it is a volume that was backed up via snapshot.

Comment (shubham-pampattiwar) [reply tests/e2e/lib/restore.go:27]: RestorePVs specifies whether to restore all included PVs from snapshot

Comment (shubham-pampattiwar) [reply tests/e2e/lib/restore.go:27]: +1 @sseago

Comment (sseago) [reply tests/e2e/lib/restore.go:27]: @shubham-pampattiwar It's confusing that the restore and backup flags aren't named similarly, but they are essentially equilalent flags, but one for backup, the other for restore.

Comment (sseago) [reply tests/e2e/lib/restore.go:27]: > The snapshot can be cloud provider based or csi snapshot. Both are applicable.

Same with SnapshotVolumes. There was a fix a while back to make this flag do the same thing for CSI and native snapshots.

Comment (shubham-pampattiwar) [reply tests/e2e/lib/restore.go:27]: yes since velero 1.12 I think, initially these 2 flags were only applicable for cloud provider based snapshots.

File: tests/e2e/backup_restore_suite_test.go
Comment (mrnold) [reply tests/e2e/backup_restore_suite_test.go:1]: Okay, I have started this process with #1345.

**Review by weshayutin — APPROVED** [review]
/LGTM

**Comment by kaovilai** [comment]
/override ci/prow/4.12-e2e-test-aws

> [ya.. the change is for a completely diff test](https://redhat-internal.slack.com/archives/C0144ECKUJ0/p1709662028012249?thread_ts=1709661802.249529&cid=C0144ECKUJ0)

:)
ps. we've seen this flake b4

**Pull request merged**
